main.c

```diff
@@ -1,10 +1,18 @@
 #define _POSIX_C_SOURCE 2
+
+#include <assert.h>
+#include <time.h>
+#include <sys/mman.h>
+#include <alloca.h>
+#include <err.h>
```

Comment:
This is a non-standard API.

Reply:
addressed

Reply:
If the mmap fails, now `exit(EXIT_FAILURE)` happens.